container/state.go

```diff
@@ -269,7 +269,7 @@ func (s *State) SetExitCode(ec int) {
 }
 
 // SetRunning sets the state of the container to "running".
-func (s *State) SetRunning(ctr libcontainerdtypes.Container, tsk libcontainerdtypes.Task, initial bool) {
+func (s *State) SetRunning(ctr libcontainerdtypes.Container, tsk libcontainerdtypes.Task, initial bool, start time.Time) {
```

Comment:
```suggestion
func (s *State) SetRunning(ctr libcontainerdtypes.Container, tsk libcontainerdtypes.Task, start *time.Time) {
```
instead of adding another parameter, perhaps it would be better to merge it with `initial` and replace it with nil check, since we won't be setting it to anything if it's false?

Reply:
@vvoland I resolved your comment by using the IsZero check and passing time.Time{} as an equivalent to nil. With nil I have to use pointers, which would get ugly quick as most calls to SetRunning are of the form SetRunning(..., Time.now), which I can't get the address of without moving Time.now out of the function.__